src/servers/buttongestureadapter.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    ButtonEvent,
+    ButtonGestureEvent,
+    CONNECT,
+    DEVICE_CHANGE,
+    DEVICE_CONNECT,
+    EVENT,
+    ROLE_BOUND,
+    SELF_ANNOUNCE,
+    SRV_BUTTON,
+    SRV_BUTTON_GESTURE,
+} from "../jdom/constants"
+import SensorServer from "./sensorserver"
+import { assert, JDDevice, JDEvent, JDService } from "../jdom/jacdac-jdom"
+import RoleManager from "./rolemanager"
+import JDServiceServer from "../jdom/serviceserver"
+
+export class AdapterServer extends JDServiceServer {
+    private roleManager? : RoleManager = undefined
+    
+    // This is a nasty hack that allows the role manager to be set after the server is instantiated.
+    // The right solution would be to have this listen for role broadcasts for something
```

Comment:
Not sure what you mean there. You can instantiate a role manager client if you want to consume one from the bus but I don't think that's what you want.

Reply:
Yeah, I basically need to get the bus role manager into the adapter server, so it can resolve the service it consumes and listen on it. This code pipes the role manager through because I don't know how do get it otherwise. RoleManagerClient seems to need a service (which doesn't seem trivial to get from a server?). Bus also seems to know about the role manager, but I don't see an event for when a server's device has its bus defined.